test/app_test.py

```diff
@@ -275,6 +275,17 @@ def f():
     assert "function(image=image)" in str(excinfo.value)
 
 
+def test_function_decorator_on_class():
+    app = App()
+    with pytest.raises(TypeError) as excinfo:
+
+        @app.function()
+        class Foo:
+            pass
+
+    assert "cannot be used on a class" in str(excinfo.value)
```

Comment:
Note you can also use the `match=` parameter of `pytest.raises` to check the message.